### ISSUE TITLE
Not able to select DMX interface

### ISSUE THREAD

**Issue by MrMdR** [nt2ds/ArtNetDMX#1]
Hi Nt2ds!
First of all; the amazing concept of your program! I'm a light artist (to be ;) ). I would just need such a program where the Art-Net data from my controller software is translated to my 'Enttec DMX USB Pro". Amazing!

When I open the program I don't seem to select the correct COM port / the Enttec device. I suppose that is the goal of the 'Uni' field?
I've read somewhere a post by you saying the program can only be used with FTDI-style interfaces, like Enttec ones. Is that correct? I didn't see it in the 'README.

I am a UI designer so if you need help designing a user-friendly interface I'm happy to help.

**Comment by nt2ds** [comment]
Hello MrMdR!
First of all thank you very much :). Secondly, the program works with all devices using the FT232RL USB to Serial chip. The difference of 'Enttec Open DMX' and 'Enttec DMX USB Pro' is that Enttec Open DMX doesn't have a microcontroller (μCU) unlike the DMX USB Pro, thus the process of converting the signal to serial is all handled by the computer (timings and such) unlike the DMX USB Pro where you just send data packets to the device with channel number and value and then the μCU of it takes care of the timing. It is the same reason the Open DMX doesn't have a data buffer and the DMX USB Pro has one. The app that I made creates a buffer on the computer so if the app is closed the enttec open dmx will have nothing to output. Now, for the DMX USB Pro, you have to send specific serial commands to the serial port that it is connected, the problem is that I don't own one and while I know how to implement it in code, there can be many bugs because of the lack of the interface itself.
To sum up, enntec DMX USB pro has a microcontroller while open dmx doesn't have one. DMX USB Pro uses different commands to output dmx because its has an μCU. I can implement it in the program but I will not be able to see how it behaves because I don't have one (DMX USB Pro).
If you want I can work on some things and send you some prototypes until I make sure it works and upload it.

**Comment by nt2ds** [comment]
Also, if you connect an Enttec Open DMX to it, and press start it will automatically connect to the enttec open dmx, no need to select the com port. That is why I didn't add such a field.

**Comment by MrMdR** [comment]
Ah, too bad the DMX USB Pro won't work at this moment.
I'm happy to help and test to make this happen! I'm no programmer myself but I do have some technical knowledge and I am very good at making screenshots, haha.
Sent a message here or email me if I can be of any service!

Some background of my use case:
I'm sending out Art-Net data from my light control software (Dot2onPC) to an internal ethernet loopback adapter . Your programme would listen to this loopback adapter and translate the Artnet to the DMX USB.

**Comment by nt2ds** [comment]
Hello again after some days, I decided now that I have time, I am able to start testing the DMX USB Pro Interface, but before that you need to do some things. I want you to connect the interface to your computer and from inside device manager send me a screenshot as to what type of device it appears (eg, COM port). I need this in order to know what type of communication (messages) my program will have to send to the device.
Thanks in advance :P
